### PR TITLE
docs: Added vue demo app

### PR DESCRIPTION
**Pull request by FuzzySid**
## Summary of change

Example app in Vue which has EmailPassword and Social Login using SuperTokens supertokens-auth-react library.

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [x] Changelog has been updated
-   ~~[ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)~~
    -   ~~Along with the associated array in `lib/ts/version.ts`~~
-   ~~[ ] Changes to the version if needed~~
    -   ~~In `package.json`~~
    -   ~~In `package-lock.json`~~
    -   ~~In `lib/ts/version.ts`~~
-   [x] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   ~~[ ] If added a new recipe interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).~~
-   ~~[ ] If I added a new recipe, I also added the recipe entry point into the `size-limit` section of `package.json` with the size limit set to the current size rounded up.~~

## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: examples/with-vue-thirdpartyemailpassword/src/router/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      path:'/auth*',
+      name:'auth',
+      component: ()=> import('../views/AuthView.vue'),
+    },
+  
+    
+   
+    
+
```
Comment (jscyo) [inline examples/with-vue-thirdpartyemailpassword/src/router/index.ts:25]: you can remove the additional newlines

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export default{
+    beforeMount(){
+        this.render()
+    },
+    beforeUpdate(){
+        this.render()
+    },
+    mounted(){
+        console.log('mounted')
```
Comment (jscyo) [inline examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue:14]: > console.log('mounted')

You should remove all console logs

**Review by jscyo — CHANGES_REQUESTED** [review]
Hey @FuzzySid , thanks for the PR, you can take a look at some of the feedback comments

**Comment by jscyo** [comment]
Also @FuzzySid , I would recommend taking a look at the `Checklist for important updates` and check the relevant items that you have done

File: examples/with-vue-thirdpartyemailpassword/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+![SuperTokens banner](https://raw.githubusercontent.com/supertokens/supertokens-logo/master/images/Artboard%20%E2%80%93%2027%402x.png)
+
+# SuperTokens ThirdpartyEmailPassword demo app with Angular and React
```
[the remark this answers is not in the document]
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/README.md]: Have fixed this in the updated PR.

File: examples/with-vue-thirdpartyemailpassword/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+### Backend
+
+-   The backend API is in the `server.ts` file.
+
+### Frontend
+
+-   The frontend code is located in the `/src` folder.
+-   Our frontend will have 2 Vue components as views, a `home` and `auth`.
+-   The `home` view omponent will use `supertokens-website` for session management and the `auth` view component will use `supertokens-auth-react` for Authentication flows and UI.
```
[the remark this answers is not in the document]
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/README.md]: Have fixed this in the updated PR.

File: examples/with-vue-thirdpartyemailpassword/src/router/index.ts (hunk 1 above)
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/src/router/index.ts]: Don't see these new lines in my editor?

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue (hunk 2 above)
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue]: Have fixed this in the updated PR.

File: examples/with-vue-thirdpartyemailpassword/src/views/HomeView.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            session: false,
+            userId: "",
+        };
+    },
+    mounted() {
+        this.getUserInfo();
+    },
+    methods: {
+        redirectToLogin() {
+            console.log("redirectToLogin called with");
```
Comment (jscyo) [inline examples/with-vue-thirdpartyemailpassword/src/views/HomeView.vue:16]: need to remove the console.log

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+export default {
+    beforeMount() {
+        this.render();
+    },
+    beforeUpdate() {
+        this.render();
+    },
+    mounted() {
+        this.render();
+    },
```
Comment (jscyo) [inline examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue:15]: Hey, I am not very familiar with Vue but do you need to load the react component in `beforeMount` and `beforeUpdate`.  We've updated the [angular example](https://github.com/supertokens/supertokens-auth-react/blob/0.23/examples/with-angular-thirdpartyemailpassword/src/app/auth/auth.component.ts#L18) also to show this, but we would like to initialize the react component only once after your the auth view component finishes initializing.

**Review by jscyo — CHANGES_REQUESTED** [review]

**Comment by jscyo** [comment]
> > Also @FuzzySid , I would recommend taking a look at the `Checklist for important updates` and check the relevant items that you have done
> 
> @jscyo Hey Joel, I have a quick query regarding this:
> 
> * For the checklist points that are not applicable with respect to the changes I have made, do I still need to tick them?

So just tick the relevant ones, for the rest just leave them or put a ~~strike through~~

File: examples/with-vue-thirdpartyemailpassword/src/router/index.ts (hunk 1 above)
Comment (jscyo) [reply examples/with-vue-thirdpartyemailpassword/src/router/index.ts]: hmm that's strange, git is picking up the new lines though

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue (hunk 2 above)
Comment (jscyo) [reply examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue]: there is one more console.log have added a comment

File: examples/with-vue-thirdpartyemailpassword/src/views/HomeView.vue (hunk 5 above)
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/src/views/HomeView.vue]: Updated the PR to reflect this change.

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue (hunk 6 above)
Comment (FuzzySid) [reply examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue]: Yeah, the `beforeMount` and `beforeUpdate` methods aren't needed. Have updated the PR to reflect that change.

**Comment by jscyo** [comment]
@FuzzySid Hey the changes look good, but you need to update the CHANGELOG.md. As mentioned in my previous comment please take a look at the contributing guide and tick the remaining checklist items when they are done

**Review by jscyo — APPROVED** [review]

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    beforeDestroy() {
+        ReactDOM.unmountComponentAtNode(document.getElementById("app") as Element);
+    },
+    methods: {
+        render() {
+            ReactDOM.render(React.createElement(SuperTokensReactComponent), document.getElementById("app"));
+        },
+    },
```
Comment (jscyo) [inline examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue:18]: You can remove the render function and just call `ReactDOM.render(React.createElement(SuperTokensReactComponent), document.getElementById("app"));` in `mounted`

**Comment by FuzzySid** [comment]
Hey @jscyo 
Running the pre-commit hook fails the compilation and I get a commit error. 
Screenshot 2022-06-27 at 9 54 44 AM
Can you suggest what I need to do to fix this?

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+-   Adds an example app with Vue
```
Comment (rishabhpoddar) [inline CHANGELOG.md:10]:
```suggestion
-   Adds an example app with Vue + React
```

File: examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import SuperTokensReactComponent from "../components/Supertokens";
+
+export default {
+    mounted() {
+        ReactDOM.render(React.createElement(SuperTokensReactComponent), document.getElementById("app"));
+    },
+    beforeDestroy() {
+        ReactDOM.unmountComponentAtNode(document.getElementById("app") as Element);
```
Comment (rishabhpoddar) [inline examples/with-vue-thirdpartyemailpassword/src/views/AuthView.vue:11]: where is "app" coming from?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Pull request merged**